### PR TITLE
Optimize quorum-k8s-hooks image

### LINKED ISSUE #140: quorum-k8s-hooks image optimization
Here are some items that I believe can be optimized in quorum-k8s-hooks Dockerfile, let me know if you are interested and I can provide PR for all/some of them.

1.  `mkdir -p /usr/share/man/man1; \` the issue   #https://github.com/balena-io-library/base-images/issues/273 seems to be fixed, no need for this anymore
2. `vim install` vim is not used anywhere in jobs
3. `helm install` helm is not used anywhere in jobs
4. `install quorum-genesis-tool during image build` hook jobs will run a bit faster
5. `add rm -rf /aws /awscliv2.zip to cleanup command` aws cli installation files, it will save about 200Mb from image size
6. `lock versions instead of using latest: debian image, kubectl, nodejs, azure-cli, awscli` currently every image build will produce different image with different tools versions, might lead to some unexpected behavior if there were breaking changes updates.

### PR DESCRIPTION
**Pull request by aliusmiles**
Fixes #140 

* remove man workaround
* remove vim install
* remove helm install
* install quorum-genesis-tool
* remove awscli install files

### CODE REVIEW

File: docker/quorum-k8s-hooks/Dockerfile [hunk 1]
```diff
@@ -2,21 +2,18 @@ FROM debian:latest
 
 RUN set -eux; \
     mkdir -p /root/.kube/ ; \
-    #https://github.com/balena-io-library/base-images/issues/273
-    mkdir -p /usr/share/man/man1; \
     apt-get update && apt-get install -y --no-install-recommends \
-    pkg-config ca-certificates apt-transport-https gnupg gnupg2 lsb-release software-properties-common jq libc6-dev make curl wget vim unzip libsodium-dev; \
+    pkg-config ca-certificates apt-transport-https gnupg gnupg2 lsb-release software-properties-common jq libc6-dev make curl wget unzip libsodium-dev; \
```
Comment (joshuafernandes) [inline docker/quorum-k8s-hooks/Dockerfile:6]: I'd like to keep vim in here to make debugging of things easier - its also not a heavyweight package

File: docker/quorum-k8s-hooks/Dockerfile [hunk 2]
```diff
@@ -2,21 +2,18 @@ FROM debian:latest
...
+    # kubectl
+    curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl" && chmod a+x ./kubectl && mv ./kubectl /usr/local/bin/kubectl ;
 
 RUN set -eux; \
     # install nodejs 17.x
     curl -fsSL https://deb.nodesource.com/setup_17.x | bash - ; \
     apt-get install -y nodejs; \
-    npm install -g npm@latest;
+    npm install -g npm@latest; \
+    npm install -g quorum-genesis-tool@latest;
```
Comment (joshuafernandes) [inline docker/quorum-k8s-hooks/Dockerfile:16]: Lock this to a version as well, so can use 0.1.6 https://www.npmjs.com/package/quorum-genesis-tool and we can update when we release new versions of the QGT

**Review by joshuafernandes — COMMENTED** [review]
Looks good @aliusmiles , couple of minor changes requested please

File: docker/quorum-k8s-hooks/Dockerfile (hunk 1 above)
Comment (aliusmiles) [reply docker/quorum-k8s-hooks/Dockerfile:6]: got it, totally makes sense.

File: docker/quorum-k8s-hooks/Dockerfile (hunk 2 above)
Comment (aliusmiles) [reply docker/quorum-k8s-hooks/Dockerfile:16]: will do

File: docker/quorum-k8s-hooks/Dockerfile [hunk 3]
```diff
@@ -1,31 +1,27 @@
...
-    #https://github.com/balena-io-library/base-images/issues/273
-    mkdir -p /usr/share/man/man1; \
-    apt-get update && apt-get install -y --no-install-recommends \
-    pkg-config ca-certificates apt-transport-https gnupg gnupg2 lsb-release software-properties-common jq libc6-dev make curl wget vim unzip libsodium-dev; \
-    echo "fd97de6b91a121428112c52e5fe04a15" > /etc/machine-id ; \
-    # kubectl    
-    curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl" && chmod a+x ./kubectl && mv ./kubectl /usr/local/bin/kubectl ; \
-    # helm
-    wget -q https://get.helm.sh/helm-v3.8.0-linux-amd64.tar.gz -O - | tar -xzO linux-amd64/helm > /usr/local/bin/helm && chmod +x /usr/local/bin/helm ;
+ARG QGT_VERSION=0.1.6
```
Comment (joshuafernandes) [inline docker/quorum-k8s-hooks/Dockerfile:3]: :+1: This is nice

**Review by joshuafernandes — APPROVED** [review]
LGTM @aliusmiles

**Pull request merged**